src/handlers/data.ts

```diff
@@ -0,0 +1,59 @@
+import { Dispatch } from "react";
+import { toast } from "react-toastify";
+import { doc, getDoc } from "firebase/firestore";
+import { db } from "../firebase.config";
+import { DataAction } from "../types";
+import { ActionType } from "../constants";
+import { isFBError } from "../types/typeguards";
+
+const getQuizHandler = async ({
+  quizId,
+  dataDispatch,
+  setIsLoading,
+}: {
+  quizId: string;
+  dataDispatch: Dispatch<DataAction>;
+  setIsLoading: Dispatch<React.SetStateAction<boolean>>;
+}) => {
+  try {
+    setIsLoading(true);
+    const response = await getDoc(doc(db, `quizes/${quizId}`));
+
+    if (response.exists()) {
+      const fbData = response.data() as {
```

Comment:
are you trying to name dbData here
if yes then change it

Reply:
no, its short for firebaseData